controllers/businessController.js

```diff
@@ -0,0 +1,189 @@
+const bcrypt = require("bcrypt");
+const sendResponse = require("../utils/response");
+const { businessSchema } = require("../models/business.model");
+var aws = require("aws-sdk");
+var multer = require("multer");
+var multerS3 = require("multer-s3");
+
+/*AWS Connection*/
+
+// aws.config.update({
+//     secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+//     accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+//     region: process.env.AWS_REGION,
+//     s3BucketEndpoint: true,
+//     endpoint: process.env.AWS_BUCKET_ENDPOINT,
+// });
+
+// var s3 = new aws.S3();
+// var upload = multer({
+//     storage: multerS3({
+//         s3: s3,
+//         acl: "public-read",
+//         bucket: process.env.AWS_BUCKET_NAME,
+//         metadata: function(req, file, cb) {
+//             cb(null, { fieldName: "business_Profile_img" });
+//         },
+//         key: function(req, file, cb) {
+//             var filename =
+//                 "public/" + Date.now().toString() + "_" + file.originalname;
+//             cb(null, filename);
+//         },
+//     }),
+// });
+// var singleUpload = upload.single("business_image");
+
+
+
+/*Register Business Profile*/
+exports.registerBusiness = async (req, res) => {
+    try{
+        var businessName = req.body.businessName;
+        var otherName = req.body.otherName;
+        var businessWebsite = req.body.businessWebsite;
+        var businessSize = req.body.businessSize;
+        var businessHQLocation = req.body.businessHQLocation;
+        var language = req.body.language;
+        var yearFounded = req.body.yearFounded;
+        var businessType = req.body.businessType;
+        
+        if (!businessName || !otherName || !businessWebsite || !businessSize || !businessHQLocation || !language || !yearFounded || !businessType) {
+            sendResponse(res, 400, 'Missing required fields');
+            return;
+        }
+        
+        const urlRegex = /^(https?:\/\/)?([\w-]+(\.[\w-]+)+)([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])?$/;
+        // Check if businessWebsite matches the URL format
+        if (!urlRegex.test(businessWebsite)) {
+            // If it doesn't match, handle the validation error
+            sendResponse(res, 400, 'Invalid businessWebsite format');
+            return;
+        }
+        
+        let isExists = await businessSchema.findOne({ businessName: req.body.businessName });
+        if (isExists) throw { message: "You have already registered" };
+        
+        let newBusiness = new businessSchema({
+            businessName: businessName,
```

Comment:
no need to write the key, value if both are same, just one time should be sufficient